assignments/lambda-classes.js

```diff
@@ -1 +1,114 @@
 // CODE here for your Lambda Classes
+class Person {
+  constructor(attributes) {
+    this.gender = attributes.gender;
+    this.age = attributes.age;
+    this.name = attributes.name;
+    this.location = attributes.location;
+  }
+  speak() {
+    return `Hello, my name is ${this.name} and Im from ${this.location}`;
+  }
+}
+
+class Instructor extends Person {
+  constructor(instructorAttrs) {
+    super(instructorAttrs);
+    this.gender = instructorAttrs.gender;
+    this.age = instructorAttrs.age;
+    this.name = instructorAttrs.name;
+    this.location = instructorAttrs.location;
+    this.isInstructor = instructorAttrs.isInstructor;
+    this.specialty = instructorAttrs.specialty;
+    this.favLanguage = instructorAttrs.favLanguage;
+    this.catchPhrase = instructorAttrs.catchPhrase;
+  }
+  demo(subject) {
+    return `Today we are learning about ${subject}`;
+  }
+  grade(subject) {
+    return `${this.name} receives a perfect score on ${subject}`;
+  }
+}
+
+class Student extends Person {
+  constructor(studentAttrs) {
+    super(studentAttrs);
+    this.gender = studentAttrs.gender;
+    this.age = studentAttrs.age;
+    this.name = studentAttrs.name;
+    this.location = studentAttrs.location;
+    this.previousBackground = studentAttrs.previousBackground;
+    this.className = studentAttrs.className;
+    this.favSubjects = studentAttrs.favSubjects;
+  }
+  listSubjects() {
+    return this.favSubjects;
+  }
+  PRAssignmnet(subject) {
+    return `${this.name} has submitted a PR for ${subject}`;
```

Comment:
Be very careful about spelling, especially in the constructor. This means that the method is misspelled for all the objects you create using this instructor.